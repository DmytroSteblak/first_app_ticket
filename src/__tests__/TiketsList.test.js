import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import TicketsList from '../components/TiketsList';

const actionState = [
    {
        price: 30546,
        carrier: 'S7',
        segments: [
            {
                origin: 'MOW',
                destination: 'HKT',
                date: '2021-04-24T02:20:00.000Z',
                stops: ['HKG', 'BKK', 'AUH'],
                duration: 1769,
            },
            {
                origin: 'HKT',
                destination: 'MOW',
                date: '2021-05-14T09:20:00.000Z',
                stops: [],
                duration: 1196,
            },
        ],
    },
]

const state = {
    searchId: '',
    tickets: [...actionState],
    loading: true,
    filterTickets: [...actionState],
}

let store;
const setUp = (state) => shallow(
    <Provider store={store}>
        <TicketsList {...state} />
    </Provider>);

describe('TicketsList Component', () => {

    it('should render component', () => {
        const mockStore = configureStore();
        store = mockStore(state);
        const ticketsList = setUp(state);
        expect(toJson(ticketsList)).toMatchSnapshot();
    });

});