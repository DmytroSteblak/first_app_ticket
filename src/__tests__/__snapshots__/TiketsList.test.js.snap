// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketsList Component should render component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <TicketsList
    filterTickets={
      Array [
        Object {
          "carrier": "S7",
          "price": 30546,
          "segments": Array [
            Object {
              "date": "2021-04-24T02:20:00.000Z",
              "destination": "HKT",
              "duration": 1769,
              "origin": "MOW",
              "stops": Array [
                "HKG",
                "BKK",
                "AUH",
              ],
            },
            Object {
              "date": "2021-05-14T09:20:00.000Z",
              "destination": "MOW",
              "duration": 1196,
              "origin": "HKT",
              "stops": Array [],
            },
          ],
        },
      ]
    }
    loading={true}
    searchId=""
    tickets={
      Array [
        Object {
          "carrier": "S7",
          "price": 30546,
          "segments": Array [
            Object {
              "date": "2021-04-24T02:20:00.000Z",
              "destination": "HKT",
              "duration": 1769,
              "origin": "MOW",
              "stops": Array [
                "HKG",
                "BKK",
                "AUH",
              ],
            },
            Object {
              "date": "2021-05-14T09:20:00.000Z",
              "destination": "MOW",
              "duration": 1196,
              "origin": "HKT",
              "stops": Array [],
            },
          ],
        },
      ]
    }
  />
</ContextProvider>
`;
